### PR TITLE
add getAvailableDevices

### LINKED ISSUE #5: Add getAvailableDevices into Core interface
It maps to [`Core::GetAvailableDevices`](https://docs.openvinotoolkit.org/latest/classInferenceEngine_1_1Core.html#acb212aa879e1234f51b845d2befae41c). The JS API sketch could be
```webidl
partial interface Core {
  sequence<DOMString> getAvailableDevices();
};
```

### PR DESCRIPTION
**Pull request by lionkunonly**
fix #5

### CODE REVIEW

**Comment by lionkunonly** [comment]
The API has been tested on windows successfully. However, I found a special sentence is printed when I tried to use the API `getAvailableDevices`.  This sentence is 
```
[E:] [BSL] found 0 ioexpander device
```
I do not print any sentence. So I think it is a log printed by the API in OpenVINO. However, the API retures a vector `[CPU  GNA GPU]` while this log says that "found 0 ioexpander device". I am puzzled.

File: src/core.cc [hunk 1]
```diff
@@ -168,4 +169,22 @@ Napi::Value Core::LoadNetwork(const Napi::CallbackInfo& info) {
   return deferred.Promise();
 }
 
+Napi::Value Core::GetAvailableDevices(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
```
Comment (huningxin) [inline src/core.cc:173]: Could you please check whether the input is valid?

File: test/core.js [hunk 2]
```diff
@@ -54,6 +54,14 @@ describe('Core Test', function() {
...
   });
 
+  it('getAvailableDevices should be a function', () => {
+    expect(core.getAvailableDevices).to.be.a('function');
+  })
+
+  it('getAvailableDevices should return a array', () => {
+    expect(core.getAvailableDevices()).to.be.a('array');
+  })
+
```
Comment (huningxin) [inline test/core.js:68]: Please add test cases for
1. valid output, like it should always contain CPU.
2. invalid input, it should raise an Error

**Review by huningxin — COMMENTED** [review]
Thanks for the PR. I've left some comments. Please take a look.

**Comment by huningxin** [comment]
> I do not print any sentence. So I think it is a log printed by the API in OpenVINO. However, the API retures a vector `[CPU GNA GPU]` while this log says that "found 0 ioexpander device". I am puzzled

I guess it might be Inference Engine internal log. If it is not in the returned vector, it would be fine.

File: src/core.cc (hunk 1 above)
Comment (lionkunonly) [reply src/core.cc:173]: Done.

File: test/core.js (hunk 2 above)
Comment (lionkunonly) [reply test/core.js:68]: Done

**Comment by huningxin** [comment]
Thanks @lionkunonly . The PR looks good to me.

**Pull request merged**